src/backend/linux_raw/mm/syscalls.rs

```diff
@@ -210,3 +219,36 @@ pub(crate) unsafe fn munlock(addr: *mut c::c_void, length: usize) -> io::Result<
 pub(crate) unsafe fn userfaultfd(flags: UserfaultfdFlags) -> io::Result<OwnedFd> {
     ret_owned_fd(syscall_readonly!(__NR_userfaultfd, flags))
 }
+
+/// Locks all pages mapped into the address space of the calling process.
+///
+/// This includes the pages of the code, data and stack segment, as well as shared libraries,
+/// user space kernel data, shared memory, and memory-mapped files. All mapped pages are
+/// guaranteed to be resident in RAM when the call returns successfully;
+/// the pages are guaranteed to stay in RAM until later unlocked.
+#[inline]
+#[cfg(any(
+    linux_kernel,
+    target_os = "freebsd",
+    target_os = "netbsd",
+    target_os = "openbsd",
+    target_os = "dragonfly",
+    target_os = "illumos",
+))]
+pub(crate) fn mlockall(flags: MlockallFlags) -> io::Result<()> {
+    unsafe { ret(syscall_readonly!(__NR_mlockall, flags)) }
```

Comment:
When used with `MCL_ONFAULT | MCL_FUTURE`, the ordering of `mlockall` with respect to arbitrary loads seems significant, because if a load happens and evokes a fault before the `mlockall`, the memory doesn't get locked, but if the load and therefore the fault happens after, then the memory does get locked. Consequently, I think this should use plain `syscall!`, rather than `syscall_readonly!`, with a comment explaining why.